testhelper/docker/resource/redis/redis.go

```diff
@@ -52,8 +59,12 @@ func Setup(ctx context.Context, pool *dockertest.Pool, d resource.Cleaner, opts
 	for _, opt := range opts {
 		opt(&conf)
 	}
+	repo := "redis"
+	if conf.repository == "" {
+		repo = conf.repository
+	}
```

Comment:
There is no need for this, you can simply follow the existing pattern that we got for `tag` and do:

```go
conf := redisConfig{
	tag:        "6",
	repository: "redis",
}
```

Reply:
This is also buggy actually. You're saying that if `conf.repository` is empty, which is the case actually if `WithRepository` is not called, then you do `repo = conf.repository` but `conf.repository` is empty. This is probably what is making the tests fail. Please fix.

Reply:
If someone was using our helper redis, they would not provide `WithRepository` option. But previously the test-case would be downloading `redis` image from docker & their tests would be passing.

If I just include changes similar to `tag`, then image would be `empty` for already existing tests & the test-cases would fail.

This the reason I included a default image